release_checklist.sh

```diff
@@ -0,0 +1,242 @@
+fi
+
+if [[ ! "$release_type" == "scheduled" && ! "$release_type" == "beta" && ! "$release_type" == "hotfix" ]]; then
+    abort "Error release type must be scheduled|beta|hotfix, you entered '$release_type'"
+fi
+
+if [[ -z "$version_number" ]]; then
+  pushd "$gb_mobile_path" >/dev/null
+
+  # Ask for new version number
+  current_version_number=$(jq '.version' package.json --raw-output)
+  #version_array=($(awk -F. '{$1=$1} 1' <<< "${current_version_number}"))
+  IFS='.' read -ar version_array <<< "$current_version_number"
+  default_version_number="${version_array[0]}.$((version_array[1] + 1)).${version_array[2]}"
+
+  read -r -p "Enter the new version number [$default_version_number]: " version_number
+  echo ""
+  version_number=${version_number:-$default_version_number}
+
+  if [[ -z "$version_number" ]]; then
+      abort "Version number cannot be empty."
+  fi
+
+  popd >/dev/null
+fi
+
+# Propmt for release date if scheduled release
+if [[ -z "$release_date" && "$release_type" == "scheduled" ]]; then
+    if command -v dateround &> /dev/null; then
+      default_release_date=$(dateround today Thurs)
+
+      read -r -p "Enter the release date (YYYY-MM-DD) [$default_release_date]: " release_date
+      release_date=${release_date:-$default_release_date}
+    else
+      read -r -p "Enter the release date (YYYY-MM-DD): " release_date
+    fi
+    echo ""
+fi
+
+checklist_template_path="$script_path/templates/release_checklist.md"
+
+pushd "$gb_mobile_path" >/dev/null
+  milestone_url=$(gh api  --method GET repos/:owner/:repo/milestones --jq ".[0].html_url")
+  default_milestone_url="https://wordpress-mobile/gutenber-mobile/milestones"
+
+  milestone_url=${milestone_url:-$default_milestone_url}
+popd >/dev/null
+
+checklist_template=$(sed -e "s/{{version_number}}/${version_number}/g" -e "s/{{release_date}}/${release_date}/g" -e "s/{{milestone_url}}/${milestone_url//\//\\/}/g" "$checklist_template_path")
+
+if [[ $release_type == "beta" || $release_type == "hotfix" ]]; then
+  release_checklist_template=$(sed "/<!-- scheduled_release_only -->/,/<!-- \/scheduled_release_only -->/d" <<< "$checklist_template")
+else
+  release_checklist_template=$(sed "/<!-- non_scheduled_release_only-->/,/<!-- \/non_scheduled_release_only -->/d" <<< "$checklist_template")
+fi
+
+if [[ -n "$include_aztec_steps" ]]; then
+  aztec_checklist_template=$(<"$script_path/templates/release_checklist_update_aztec.md")
+  aztec_checklist_template=${aztec_checklist_template//$'\n'/\\n}
+  release_checklist_template=$(sed -e "s/.*optional_aztec_release.*/$aztec_checklist_template/" <<< "$release_checklist_template")
+fi
+
+issue_title="Release checklist for v$version_number ($release_type)"
+issue_label="release checklist,$release_type release"
+issue_assignee="@me"
+issue_body=$'# Release Checklist\n'"This checklist is for the $release_type release v$version_number."$'\n\n **Release date:** '"$release_date"$'\n'"$checklist_message"$'\n'"$release_checklist_template"
```

Comment:
We don't provide a release date in beta and hotfixes so we should not include it here.